xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -386,27 +464,60 @@ void CAESinkAUDIOTRACK::GetDelay(AEDelayStatus& status)
   // In their infinite wisdom, Google decided to make getPlaybackHeadPosition
   // return a 32bit "int" that you should "interpret as unsigned."  As such,
   // for wrap saftey, we need to do all ops on it in 32bit integer math.
+
   uint32_t head_pos = (uint32_t)m_at_jni->getPlaybackHeadPosition();
 
-  double delay;
-  delay = m_duration_written - ((double)head_pos / m_sink_sampleRate);
+  // head_pos does not necessarily start at the beginning
+  if (m_offset == -1 && m_at_jni->getPlayState() == CJNIAudioTrack::PLAYSTATE_PLAYING)
+  {
+    CLog::Log(LOGDEBUG, "Offset updated to %u", head_pos);
+    m_offset = head_pos;
+  }
 
-  m_smoothedDelayVec.push_back(delay);
-  if (m_smoothedDelayCount <= SMOOTHED_DELAY_MAX)
-    m_smoothedDelayCount++;
-  else
-    m_smoothedDelayVec.erase(m_smoothedDelayVec.begin());
+  if (m_offset > head_pos)
+  {
+    CLog::Log(LOGDEBUG, "You did it wrong man - fully wrong! offset %lld head pos %u", m_offset, head_pos);
+    m_offset = 0;
+  }
+  uint32_t normHead_pos = head_pos - m_offset;
 
-  double smootheDelay = 0;
-  for (double d : m_smoothedDelayVec)
-    smootheDelay += d;
-  smootheDelay /= m_smoothedDelayCount;
+  if (m_passthrough && !m_info.m_wantsIECPassthrough)
+  {
+    if (m_pause_counter > 0)
+    {
+      const double d = GetMovingAverageDelay(GetCacheTotal());
+      CLog::Log(LOGDEBUG, "Faking Delay: smooth %lf measured: %lf", d * 1000, GetCacheTotal() * 1000);
+      status.SetDelay(d);
+      return;
+    }
+  }
+  if (normHead_pos > m_lastPlaybackHeadPosition)
+  {
+    unsigned int differencehead = normHead_pos - m_lastPlaybackHeadPosition;
+    CLog::Log(LOGDEBUG, "Sink advanced: %u", differencehead);
+    m_lastPlaybackHeadPosition = normHead_pos;
+  }
 
-#ifdef DEBUG_VERBOSE
-  CLog::Log(LOGDEBUG, "CAESinkAUDIOTRACK::GetDelay m_frames_written/head_pos %f/%u %f", m_duration_written, head_pos, smootheDelay);
-#endif
+  double gone = (double)normHead_pos / (double) m_sink_sampleRate;
```

Comment:
I don't care either way as long as it's consistent. But it's really just nit picking - so feel free to ignore :)